### PR TITLE
Issue #776 Some more javadoc fixes

### PR DESCRIPTION
**Pull request by ljacomet**
* `Store` javadoc
* Moved `org.ehcache.function` to `org.ehcache.core.spi.function`
* Misc fixes

### CODE REVIEW

File: core/src/main/java/org/ehcache/core/internal/util/Functions.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 public class Functions {
 
   /**
-   * Will transform the passed in {@link org.ehcache.function.Function} in a apply once and only once Function.
+   * Will transform the passed in {@link Function} in a apply once and only once Function.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/internal/util/Functions.java]: "in to an apply once..."

File: core/src/main/java/org/ehcache/core/internal/util/Functions.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
   }
 
   /**
-   * Will transform the passed in {@link org.ehcache.function.BiFunction} in a apply once and only once BiFunction.
+   * Will transform the passed in {@link BiFunction} in a apply once and only once BiFunction.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/internal/util/Functions.java]: ditto

File: core/src/main/java/org/ehcache/core/spi/cache/Store.java [hunk 3]
```diff
@@ -34,12 +34,16 @@
 import java.util.concurrent.TimeUnit;
 
 /**
- * The Service Provider Interface is what a {@link org.ehcache.Cache Cache} instance requires to be able to store
- * Cache entries (i.e. mappings of key to value, including all metadata).
- * It is basically a {@link java.util.concurrent.ConcurrentMap} with built in eviction/expiration. Possibly, it represents a view
- * on data held on some persistent and/or remote storage.
+ * The {@code Store} interface represents the backing storage of a {@link Cache}. It abstracts the support for multiple
+ * tiers and eventing, eviction and expiry concerns.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/cache/Store.java]: weird list wording

File: core/src/main/java/org/ehcache/core/spi/cache/Store.java [hunk 4]
```diff
@@ -34,12 +34,16 @@
...
+ * <P>
+ *   It maps keys of type K to {@link ValueHolder value holder} which contains value V and associated metadata. As such
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/cache/Store.java]: weird plurals

File: core/src/main/java/org/ehcache/core/spi/cache/Store.java [hunk 5]
```diff
@@ -181,20 +221,27 @@
...
    * @throws CacheAccessException if the mapping can't be replaced
    */
   ReplaceStatus replace(K key, V oldValue, V newValue) throws CacheAccessException;
 
   /**
-   * Removes all of the mappings from this map.
-   * This method provides no guarantee in terms of atomicity.
+   * Removes all of the mappings from this {@code Store}.
+   * <P>
+   * This method provides no guarantee in terms of atomicity in the face of concurrent operations.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/cache/Store.java]: I think "in terms" is redundant here. "This method provides no guarantee of atomicity in the face of concurrent operations" - although maybe even the last half is unnecessary too.

File: core/src/main/java/org/ehcache/core/spi/cache/Store.java [hunk 6]
```diff
@@ -208,9 +255,11 @@
   StoreEventSource<K, V> getStoreEventSource();
 
   /**
-   * Returns an iterator over the elements in this store.  The elements are
-   * returned in no particular order (unless this set is an instance of some
+   * Returns an iterator over the elements in this store.
+   * <P>
+   *   The elements are returned in no particular order (unless this set is an instance of some
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/cache/Store.java]: Whats the point of the "unless..." clause?  "Instances of this interface behave as document unless documented to behave differently"?

File: core/src/main/java/org/ehcache/core/spi/cache/Store.java [hunk 7]
```diff
@@ -348,16 +569,20 @@
   /**
    * Holds both a value, and all the metadata associated with a mapping in a Store.
    *
-   * @param <V>
+   * @param <V> the value type
    */
   interface ValueHolder<V> {
+
+    /**
+     * Constant value indicating no expiration or eternal mapping.
```
Comment (chrisdennis) [inline core/src/main/java/org/ehcache/core/spi/cache/Store.java]: "(an eternal mapping)"

File: core/src/main/java/org/ehcache/core/spi/cache/Store.java (hunk 6 above)
Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/spi/cache/Store.java]: That's javadoc copy paste, getting rid of it.

**Pull request merged**